salt/modules/vsphere.py

```diff
@@ -2530,11 +2554,12 @@ def list_non_ssds(host, username, password, protocol=None, port=None, host_names
         salt '*' vsphere.list_non_ssds my.vcenter.location root bad-password \
         host_names='[esxi-1.host.com, esxi-2.host.com]'
     '''
-    service_instance = salt.utils.vmware.get_service_instance(host=host,
-                                                              username=username,
-                                                              password=password,
-                                                              protocol=protocol,
-                                                              port=port)
+    if any([username, password, protocol, port]):
+        service_instance = salt.utils.vmware.get_service_instance(host=host,
+                                                                  username=username,
+                                                                  password=password,
+                                                                  protocol=protocol,
+                                                                  port=port)
```

Comment:
I guess all of these. Actually, would it make sense to just extract this behavior to a `def _get_service_instance(host, username, password, protocol, port, default=service_instance)` (or whatever)?